src/circuits/etable_compact/op_configure/op_bin_bit.rs

```diff
@@ -0,0 +1,280 @@
+use super::*;
+use crate::{
+    circuits::{
+        mtable_compact::encode::MemoryTableLookupEncode,
+        rtable::{BinOp, pow_table_encode},
+        utils::{bn_to_field, Context},
+    },
+    constant,
+};
+use halo2_proofs::{
+    arithmetic::FieldExt,
+    plonk::{Error, Expression, VirtualCells},
+};
+use specs::itable::{BitOp, OPCODE_ARG1_SHIFT};
+use specs::mtable::VarType;
+use specs::step::StepInfo;
+use specs::{
+    etable::EventTableEntry,
+    itable::{OpcodeClass, OPCODE_ARG0_SHIFT, OPCODE_CLASS_SHIFT},
+};
+
+
+pub struct BinBitConfig {
+    lhs: U64Cell,
+    rhs: U64Cell,
+    res: U64Cell,
+    op: U4BopCell,
+    op_class: UnlimitedCell,
+    op_lookup: PowTableLookupCell,
+    vtype: CommonRangeCell,
+
+    lookup_stack_read_lhs: MTableLookupCell,
+    lookup_stack_read_rhs: MTableLookupCell,
+    lookup_stack_write: MTableLookupCell,
+}
+
+
+pub struct BinBitConfigBuilder {}
+
+
+impl<F: FieldExt> EventTableOpcodeConfigBuilder<F> for BinBitConfigBuilder {
+    fn configure(
+        common: &mut EventTableCellAllocator<F>,
+        constraint_builder: &mut ConstraintBuilder<F>,
+    ) -> Box<dyn EventTableOpcodeConfig<F>> {
+        let lhs = common.alloc_u64();
+        let rhs = common.alloc_u64();
+        let res = common.alloc_u64();
+        let op = common.alloc_u4_bop();
+        let op_class = common.alloc_unlimited_value();
+        let op_lookup = common.alloc_pow_table_lookup();
+        let vtype = common.alloc_common_range_value();
+
+        let lookup_stack_read_lhs = common.alloc_mtable_lookup();
+        let lookup_stack_read_rhs = common.alloc_mtable_lookup();
+        let lookup_stack_write = common.alloc_mtable_lookup();
+
+        constraint_builder.push(
+            "binbit op class",
+            Box::new(move |meta| {
+                vec![op_lookup.expr(meta) - pow_table_encode(op.expr(meta), constant_from!(12) * op_class.expr(meta))]
+            })
+        ); 
+
+        constraint_builder.push(
+            "binbit eq constraints",
+            Box::new(move |meta| {
+                vec![
+                    op.eq_constraint(meta)
```

Comment:
It is correct, but using square limits max of op number. Just sum of the op, and verify is a 16x of op.

Reply:
good idea!